### PR TITLE
[10.x.x] URP XRSDK improvements

### PR DESCRIPTION
**Pull request by thomas-zeng**
# Purpose of this PR
This is the follow-up PR to PR#22 to address fix-in-next-pr issues
Detailed items are tracked here: https://jira.unity3d.com/browse/XRE-2580

# Testing status
## Manual Tests
- UniversalRenderingExample

## Automated Tests
- ABV

## Links
**Yamato**: (Select your branch) https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

# Comments to reviewers

This is the followup PR to PR#22 to address fix-in-next-pr issues:
* Moved stereo shader properties to UniversalRenderPipelineCore
* Added XR ifdef to xr asset in ForwardRendererData.cs
* Fixed the MSAA on Quest when camera has msaa disabled in camera inspector

To QA:
* Please verify on Quest when camera has msaa disabled, camera setting will override URP's pipeline asset msaa settings

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 1]
```diff
@@ -256,8 +256,8 @@ private void DrawFullscreenMesh(CommandBuffer cmd, Material material, int passIn
         {
             if (m_UseDrawProcedural)
             {
-                Vector4 scaleBias = new Vector4(1, 1, 0, 0);
-                cmd.SetGlobalVector(ShaderPropertyId.scaleBias, scaleBias);
+                Vector4 scaleBiasSrcUV = new Vector4(1, 1, 0, 0);
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:259]: for src uv can't we use UV_STARTS_AT_TOP and/or projectionParams? 

https://docs.unity3d.com/Manual/SL-PlatformDifferences.html

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs [hunk 2]
```diff
@@ -103,13 +103,13 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
...
-            // scaleBias.z = bias
-            // scaleBias.w = unused
+            // scaleBiasDstUV.x = flipSign
+            // scaleBiasDstUV.y = scale
+            // scaleBiasDstUV.z = bias
+            // scaleBiasDstUV.w = unused
             float flipSign = (cameraData.IsCameraProjectionMatrixFlipped()) ? -1.0f : 1.0f;
-            Vector4 scaleBiasRt = (flipSign < 0.0f) ? new Vector4(flipSign, 1.0f, -1.0f, 1.0f) : new Vector4(flipSign, 0.0f, 1.0f, 1.0f);
-            cmd.SetGlobalVector(ShaderPropertyId.scaleBiasRt, scaleBiasRt);
+            Vector4 scaleBiasDstUV = (flipSign < 0.0f) ? new Vector4(flipSign, 1.0f, -1.0f, 1.0f) : new Vector4(flipSign, 0.0f, 1.0f, 1.0f);
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:111]: maybe we should have a helper method for this? What is the workflow for users that want to do blit in URP? F.ex, now the 2nd most voted card and feedback is about custom post-processing.

Seems like we need a Blit/DrawFullscreen helper here to encapsulate this scale bias logic.

**Comment by phi-lira** [comment]
Adding @ellioman to this PR as he's working on SSAO and might have feedback about blitting API.

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs (hunk 1 above)
Comment (thomas-zeng) [reply com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:259]: I will investigate this tomorrow.

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs (hunk 2 above)
Comment (thomas-zeng) [reply com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:111]: Thanks felipe. I agree we should use helper function. I will look into this.

**Comment by phi-lira** [comment]
@thomas-zeng what's status on this PR?

**Comment by thomas-zeng** [comment]
PR requires further tuning. Will reopen once it is ready.

**Pull request closed without merging**